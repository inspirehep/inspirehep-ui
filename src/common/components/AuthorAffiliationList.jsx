import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { List } from 'immutable';

import ExternalLink from './ExternalLink';
import InlineList from './InlineList';

class AuthorAffiliationList extends Component {
  static renderAffiliation(affiliation) {
    const affiliationHref = `//inspirehep.net/search?cc=Institutions&p=institution:"${affiliation}"`;
    return (
      <ExternalLink className="secondary-link" href={affiliationHref}>
        {affiliation}
      </ExternalLink>
    );
  }

  render() {
    const { affiliations } = this.props;
    if (affiliations) {
      return (
        <InlineList
          wrapperClassName="di"
          separateItemsClassName="separate-items-with-and"
          items={affiliations}
          renderItem={AuthorAffiliationList.renderAffiliation}
        />
      );
    }
    return null;
  }
}

AuthorAffiliationList.propTypes = {
  affiliations: PropTypes.instanceOf(List),
};

AuthorAffiliationList.defaultProps = {
  affiliations: null,
};

export default AuthorAffiliationList;
