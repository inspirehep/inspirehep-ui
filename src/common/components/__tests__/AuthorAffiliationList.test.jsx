import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import AuthorAffiliationList from '../AuthorAffiliationList';

describe('AuthorAffiliationList', () => {
  it('renders author with one affiliation', () => {
    const affiliations = fromJS(['Affiliation']);
    const wrapper = shallow(
      <AuthorAffiliationList affiliations={affiliations} />
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('renders author with multiple affiliations', () => {
    const affiliations = fromJS(['Affiliation1', 'Affiliation2']);
    const wrapper = shallow(
      <AuthorAffiliationList affiliations={affiliations} />
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
