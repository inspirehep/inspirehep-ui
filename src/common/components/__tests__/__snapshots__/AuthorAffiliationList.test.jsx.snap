// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorAffiliationList renders author with multiple affiliations 1`] = `
<div
  className="__InlineList__ di"
>
  <ul
    className="separate-items-with-and"
  >
    <li
      key="Map { \\"value\\": \\"Affiliation1\\" }"
    >
      <ExternalLink
        className="secondary-link"
        href="//inspirehep.net/search?cc=Institutions&p=institution:\\"Affiliation1\\""
      >
        Affiliation1
      </ExternalLink>
    </li>
    <li
      key="Map { \\"value\\": \\"Affiliation2\\" }"
    >
      <ExternalLink
        className="secondary-link"
        href="//inspirehep.net/search?cc=Institutions&p=institution:\\"Affiliation2\\""
      >
        Affiliation2
      </ExternalLink>
    </li>
  </ul>
</div>
`;

exports[`AuthorAffiliationList renders author with one affiliation 1`] = `
<div
  className="__InlineList__ di"
>
  <ul
    className="separate-items-with-and"
  >
    <li
      key="Map { \\"value\\": \\"Affiliation\\" }"
    >
      <ExternalLink
        className="secondary-link"
        href="//inspirehep.net/search?cc=Institutions&p=institution:\\"Affiliation\\""
      >
        Affiliation
      </ExternalLink>
    </li>
  </ul>
</div>
`;
